### ISSUE TITLE
Pad operation gradient crashes with Metal abort

### ISSUE THREAD

**Issue by tillahoffmann** [tillahoffmann/jax-mps#59]
## Description

The `stablehlo.pad` operation works correctly for the forward pass, but computing gradients causes a fatal Metal abort with a shape mismatch error in `sliceUpdateDataTensor`.

This affects both edge-only padding and interior padding (added in PR #50).

## Error

```
sliceUpdateDataTensor shape mismatch
```

(Fatal Metal abort - not a catchable exception)

## Impact

- Pad operation gradients are disabled (`differentiable_argnums=()`)
- Does **not** block pooling gradients (PR #48) since pooling's backward pass only needs pad's forward pass

## To reproduce

```python
import jax
import jax.numpy as jnp

def f(x):
    return jax.lax.pad(x, 0.0, [(1, 1, 0), (1, 1, 0)])

x = jnp.ones((3, 3))
jax.grad(lambda x: f(x).sum())(x)  # Crashes
```

## Notes

This appears to be an issue with how MPS Graph handles the backward pass for strided slice updates. May require a different approach for the gradient computation.
